src/hsl_mc66.jl

```diff
@@ -0,0 +1,100 @@
+export mc66, mc66_control
+
+mutable struct fa14_seed
+  ix::Int32
+end
+
+fa14_seed() = fa14_seed(65535)
+
+mutable struct mc66_control{T <: BLAS.BlasReal}
+  lp::Int32
+  wp::Int32
+  mp::Int32
+  print_level::Int32
+  kl_aggressive::T
+  max_imbalance::T
+  mglevel::Int32
+  grid_rdc_fac::T
+  coarsest_size::Int32
+  coarsen_scheme::Int32
+  num_coarsest_kl::Int32
+end
+
+function mc66_control{T}(; lp=6, wp=6, mp=6, print_level=-1, kl_aggressive=-1, max_imbalance=0.01,
+                           mglevel=typemax(Int32), grid_rdc_fac=0.75, coarsest_size=100,
+                           coarsen_scheme=1, num_coarsest_kl=4) where {T <: BLAS.BlasReal}
+
+  control = mc66_control{T}(lp, wp, mp, print_level, kl_aggressive, max_imbalance, mglevel,
+                            grid_rdc_fac, coarsest_size, coarsen_scheme, num_coarsest_kl)
+  return control
+end
+
+# Thanks to `nm -D libhsl_mc66.so`!
+function mc66s(m, n, nz, irn, jcn, nblocks, control, seed, row_order, info, rowptr, column_order, colptr, netcut, rowdiff, kblocks)
+  ccall((:__hsl_mc66_simple_MOD_monet, libhsl_mc66),
+        Cvoid,
+        (Ref{Int32}, Ref{Int32}, Ref{Int32}, Ptr{Int32}, Ptr{Int32}, Ref{Int32}, Ref{mc66_control{Float32}}, Ref{fa14_seed}, Ptr{Int32}, Ref{Int32}, Ptr{Int32}, Ptr{Int32}  , Ptr{Int32}, Ref{Int32}, Ref{Float32}, Ref{Int32}),
+         m         , n         , nz        , irn       , jcn       , nblocks   , control                   , seed          , row_order , info      , rowptr    , column_order, colptr    , netcut    , rowdiff     , kblocks   )
+end
+
+function mc66s_print_message(info)
+  lp = Ref{Int32}(6)
+  wp = Ref{Int32}(6)
+  context = Ptr{Cvoid}()
+  ccall((:__hsl_mc66_single_MOD_monet_print_message, libhsl_mc66),
```

Comment:
I have a segmentation fault with this `ccall`.
Maybe the mangling is different... It's the first time that I wrapped a routine from a Fortran submodule.